crates/matrix-sdk-crypto/src/identities/device.rs

```diff
@@ -158,11 +158,81 @@ impl Device {
     /// encrypted `m.room_key` event. This method determines if this `Device`
     /// can be confirmed as the creator and owner of the `m.room_key`.
     pub fn is_owner_of_session(&self, session: &InboundGroupSession) -> bool {
+        // An imported room key means that we did not receive the room key as a
+        // `m.room_key` event when the room key was initially exchanged.
+        //
+        // This could mean a couple of things:
+        //      1. We received the room key as a `m.forwarded_room_key`
+        //      2. We imported the room key through a file export.
+        //      3. We imoprted the room key through a backup.
+        //
+        // To be certain that a `Device` is the owner of a room key we need to have a
+        // proof that the `Curve25519` key of this `Device` was used to
+        // initially exchange the room key. This proof is provided by the Olm decryption
+        // step, see bellow for further clarification.
+        //
+        // Each of the above room key methods that receive room keys do not contain this
+        // proof and we received only a claim that the room key is tied to a
+        // `Curve25519` key.
+        //
+        // Since there's no way to verify that the claim is true, we say that we don't
+        // know that the room key belongs to this device.
         if session.has_been_imported() {
             false
         } else if let Some(SigningKey::Ed25519(key)) =
             session.signing_keys.get(&DeviceKeyAlgorithm::Ed25519)
         {
+            // Room keys are received as an `m.room.encrypted` event using the `m.olm`
+            // algorithm. Upon decryption of the `m.room.encrypted` event, the
+            // decrypted content will contain also a `Ed25519` public key.
+            //
+            // The inclusion of this key means that the `Curve25519` key of the `Device` and
+            // Olm `Session`, established using the DH authentication of the
+            // double ratchet, is pointing at the `Ed25519` key of the `Device`
```

Comment:
what does it mean for a curve25519 key to "point at" an ed25519 key?

Reply:
This wording sounded suboptimal to me too since it's undefined jargon. I think it would be better to say "binds" or "cryptographically binds" since cryptographic binding should be a well-understood term.

Reply:
It's just an euphemism for "I claim that this Ed25519 key is part of my identity", same way the Ed25519 key claims the same via the signature. Anybody could create an Ed25519 key and pick up any public Curve25519 key from the server and claim the same.

The thing only works if both claims, or pointers, match up.

Reply:
> This wording sounded suboptimal to me too since it's undefined jargon. I think it would be better to say "binds" or "cryptographically binds" since cryptographic binding should be a well-understood term.

Sure, let's change it to "binds".

Reply:
Did so in https://github.com/matrix-org/matrix-rust-sdk/pull/1371/commits/4e5c9c4a4881670a70fc6592717dd37d16538e60.